### DISCUSSION TITLE
Support and Resistance

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#217] [Ideas]
@abcdamir2 had brought up [Support and Resistance](https://school.stockcharts.com/doku.php?id=chart_analysis:support_and_resistance) as an indicator (#214).  This is something I've had in [the backlog](https://github.com/DaveSkender/Stock.Indicators/projects/1#card-50260249) for a while, but have yet to find a standard method for computing it as an actual indicator in the library.  Any thoughts or sources on how we might do it?

**Comment by M22arius494** [comment]
I like the idea! Isn't that more or less the same as the ZigZag indicator? Not sure..
Overlapping perhaps.

**Reply by DaveSkender** [reply]
Yes, that's actually why I started with [ZigZag](https://daveskender.github.io/Stock.Indicators/indicators/ZigZag/#content) as I thought it might be an intermediate step to support and resistance; or, as you say, can be used as _the_ indicator for support and resistance.  The only problem I have with ZigZag is that it lacks any consideration for the trading volume.

![image](https://user-images.githubusercontent.com/8432125/101672719-39546d00-3a24-11eb-8072-9410ffdc77f0.png)

**Reply by DaveSkender** [reply]
The other options are [Donchian Channel](https://daveskender.github.io/Stock.Indicators/indicators/Donchian/#content) and [Pivot Points](https://school.stockcharts.com/doku.php?id=technical_indicators:pivot_points).  I've never tried to use Pivot Points.  Too confusing IMO.  We haven't implemented Pivot Points yet, but have it low in the backlog.

![image](https://user-images.githubusercontent.com/8432125/101673396-1c6c6980-3a25-11eb-83b4-d58e97393390.png)

**Reply by M22arius494** [reply]
> Yes, that's actually why I started with [ZigZag](https://daveskender.github.io/Stock.Indicators/indicators/ZigZag/#content) as I thought it might be an intermediate step to support and resistance; or, as you say, can be used as _the_ indicator for support and resistance. The only problem I have with ZigZag is that it lacks any consideration for the trading volume.
> 
> ![image](https://user-images.githubusercontent.com/8432125/101672719-39546d00-3a24-11eb-8072-9410ffdc77f0.png)

I tried to implement Peaks and troughs trend prediction with it, but i got stuck on comparing tops and bottoms (need some more time). Also no idea what percentage are suitable for each interval.

**Reply by DaveSkender** [reply]
It's tough.  How do you get an algorithm to find what seems to be obvious support and resistance lines?
Maybe it's easy.  I'm hoping there are reputable and reliable methods out there somewhere.

![image](https://user-images.githubusercontent.com/8432125/101696085-8dbb1500-3a43-11eb-9466-91e120c32dc6.png)

**Reply by aemamj** [reply]
sorry i answer late and sorry about my language . i think [Fractal](https://stockindicators.dev/indicators/Fractal/#content) is better than zigzag . you could list of point from fractal then which on has more repeat is best of support or Resistance . and if the point bearish fractal or bullish fractal has a long time is better .

**Reply by GmiSoft** [reply]
Hi, 
First of all, thank you for this library 🥇  👍 .
I'm new to trading using technical analysis so I'm sorry if I'm off the mark.
What about using Pivots points PP R1 R2 R3 and S1 S2 S3.
There is a standard calculation and some other popular like Fibonacci and Woodie's.
PS : Sorry about my language also

**Reply by DaveSkender** [reply]
Thank you for some thoughts on this.  Let me break these three items out into separate threads to discuss further.

**Reply by DaveSkender** [reply]
I was just randomly looking at my [chart](https://charts.stockindicators.dev) and had a thought that it'd be fairly easy to add these trailing support and resistance lines to _Williams Fractal_.  I think @aemamj is on the right track.

![image](https://github.com/DaveSkender/Stock.Indicators/assets/8432125/4e97b478-07b2-4f8a-8857-878e07dce0d5)

**Comment by DaveSkender** [comment]
**Fibonacci Retracement Levels**: anybody use them successfully?  I've never quite understood the logic behind them; other than that they claim to represent "natural" patterns in price movement.

I loved the DaVinci Code (movie) with Nick Cage.  It introduced me to Fibonacci numbers.  Later, I used Fibonacci numbers in Planning Poker with Scrum teams.

How it can be used to predict price behavior or support and resistance, is a mystery to me, like the DaVinci Code.  Isn't it like drawing an arbitrary line in the sand and saying that's supposed to mean something?

Fibonacci is also used in one of [several forms of Pivot Points](https://www.babypips.com/learn/forex/other-pivot-point-calculation-methods).

**Reply by DaveSkender** [reply]
Since Pivot Points in general seems to be a legitimate indicator, I'll move it up [in the backlog](https://github.com/DaveSkender/Stock.Indicators/projects/1#card-50261046) for v1.  Hopefully along the way I can figure out how to use them correctly (a learning lesson for me).  I'll likely have a parameter to specify which variant to return, rather than creating separate methods for each flavor.

**Reply by DaveSkender** [reply]
[Pivot Points](https://daveskender.github.io/Stock.Indicators/indicators/PivotPoints/#content) is now available in version 1.4.0

**Comment by DaveSkender** [comment]
Was just checking out Barcharts.com.  They have an interesting page enumerating different levels of support and resistance:
Example: https://www.barchart.com/etfs-funds/quotes/SPY/cheat-sheet

**Comment by metathron** [comment]
This might be interessting.
[https://www.quanttrader.com/index.php/detecting-support-and-resistance-levels/KahlerPhilipp2020](https://www.quanttrader.com/index.php/detecting-support-and-resistance-levels/KahlerPhilipp2020)

**Reply by DaveSkender** [reply]
That is a interesting idea.  Looks like he’s using some calculus on EMA to then back-identify the last high/low.  It’s got a lag due to the EMA, but I suspect most support / resistance indicators do.  Might work faster with HMA or one of the newer moving averages.

**Comment by aemamj** [comment]
i read any document these days that to find a way to show "Support and Resistance" . when I have read a pdf for fuzzy algorithms i saw a indicator in the pdf that name is "Support and Resistance" . this [pdf ](https://www.researchgate.net/publication/288441776_Fuzzy_and_Fractal_Technology_in_Market_Analysis )is

**Reply by DaveSkender** [reply]
Their equation for Support and Resistance is  [Bollinger Bands](https://daveskender.github.io/Stock.Indicators/indicators/BollingerBands/#content), right?  I think it is.  See page 6.

![image](https://user-images.githubusercontent.com/8432125/102839135-d4a6f400-43cd-11eb-8a95-988b9b5a7c68.png)

**Reply by DaveSkender** [reply]
According to the authors of that paper:

> Conclusion: The result is that it is not possible to forecast market change ...

I guess I’ll just quit and find a new hobby! 😄

**Comment by preintel** [comment]
Wanted to ask if there is anyone that was able to figure out determining support and resistance lines using a combination of the pre-existing technical indicators?
Thinking if we could use slope/linear regressions on close prices to smooth out the data/noise then determine inflection points on the curve using ?Williams Fractals then determining the slope between consecutive fractals. Although this will probably yield something more like a price channel/trend lines rather than horizontally directed support/resistance lines. I suppose we could determine inflection points and compare them to see how close they are in terms of y value to each other to draw horizontal (y=constant) lines. Something like pivot points but more consistent across the x-axis.

A thread on possible solutions: https://stackoverflow.com/questions/8587047/support-resistance-algorithm-technical-analysis

**Reply by preintel** [reply]
Or maybe machine learning?
https://medium.com/swlh/using-machine-learning-to-locate-support-and-resistance-lines-for-stocks-461d590e9925

**Comment by g5becks** [comment]
https://www.tradingview.com/script/Ej53t8Wv-Support-Resistance-Channels/

This is my favorite support and resistance indicator and it’s based solely on pivot points.
It creates a channel based on the history of pivots.

**Reply by DaveSkender** [reply]
Solid use of [Pivots](https://dotnet.stockindicators.dev/indicators/Pivots/#content).  Like.
